brax/training/agents/ppo/train.py

```diff
@@ -231,12 +231,16 @@ def train(
   key_envs = jnp.reshape(key_envs,
                          (local_devices_to_use, -1) + key_envs.shape[1:])
   env_state = reset_fn(key_envs)
+  ndarray_obs = isinstance(env_state.obs, jnp.ndarray) # Check whether observations are in dictionary form.
+
+  obs_shape = env_state.obs.shape[-1] if ndarray_obs \
+    else jax.tree_util.tree_map(lambda x: x.shape[2:], env_state.obs) # Discard batch axes.
 
   normalize = lambda x, y: x
   if normalize_observations:
     normalize = running_statistics.normalize
   ppo_network = network_factory(
```

Comment:
Don't we need to update the `network_factory` impl as well? Had an earlier comment along these lines

Reply:
Thanks!